custom_components/robovac_mqtt/vacuum.py

```diff
@@ -36,48 +34,58 @@ async def async_setup_entry(
         device = await eufy_clean.init_device(vacuum['deviceId'])
         await device.connect()
         _LOGGER.info("Adding vacuum %s", device.device_id)
-        entity = RoboVacMQTTEntity(device)
+        entity = RoboVacMQTTEntity(device, hass)
         hass.data[DOMAIN][VACS][device.device_id] = entity
         async_add_entities([entity])
+
+        battery_sensor = RobovacBatterySensor(device)
+        async_add_entities([battery_sensor])
+
         await entity.pushed_update_handler()
 
+class RobovacBatterySensor(Entity):
```

Comment:
@CBDesignS what is the purpose of this battery sensor? I just now realized that there is a battery attribute in the vacuum already.

Reply:
e.g by this template I can get my vacuums current battery level

```py
{{state_attr('vacuum.robovac_x10_pro_omni', 'battery_level')}}
```